.github/workflows/run-unit-tests.yml

```diff
@@ -83,16 +83,23 @@ jobs:
           fi
           rm -rf ~/.colima ~/.lima
           brew install --HEAD colima
-          brew services start colima
           brew install docker
-          colima delete
-          # Attempt to start Colima
-          ATTEMPT_LIMIT=3
+
+          export DOCKER_CONTEXT="colima-$GITHUB_RUN_ID"
+          docker context create $DOCKER_CONTEXT --docker host=unix:///var/run/docker.sock
 
           start_colima() {
-            colima start --network-address --arch x86_64 --cpu=1 --memory=1
+            # Find a free port
+            RANDOM_PORT=$((RANDOM % 16384 + 49152))
+
+            echo "Using random port: $RANDOM_PORT for SSH"
+
+            colima start --network-address --arch x86_64 --cpu=1 --memory=1 --ssh-port $RANDOM_PORT
+            docker context use $DOCKER_CONTEXT
```

Comment:
Also. this line changes the exit status of this function. Source: [man bash](https://www.man7.org/linux/man-pages/man1/bash.1.html#:~:text=When%20executed%2C%20the%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20exit%20status%20of%20a%20function%20is%20the%20exit%20status%20of%20the%20last%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20command%20executed%20in%20the%20body.)